micromagneticdata/drive.py

```diff
@@ -415,3 +416,77 @@ def slider(self, description="step", **kwargs):
+        ``xarray.DataArray`` from it.
+
+        Depending on ``micromagneticdata.Drive.info['driver']``, the dimensions and
+        coordinates of the output changes. For ``MinDriver``, which only contains a
+        single ``discretisedfield.Field``, the value of the single
+        ``discretisedfield.Field.to_xarray()`` is returned. For ``TimeDriver``, the
+        output contains an extra dimension ``t`` which corresponds to the time steps of
+        the simulation and is obtained by concatenating individual
+        ``discretisedfield.Field.to_xarray()`` in the ``micromagneticdata.Drive``. The
+        coordinates of ``t`` give the values of time at given time steps. Similarly,
+        for ``HysteresisDriver``, the extra dimension is ``B_hysteresis``. It has four
+        coordinates, namely ``B_hysteresis``, ``Bx_hysteresis``, ``By_hysteresis``, and
+        ``Bz_hysteresis``. The first represents the norm of the hysteresis field, while
+        the rest three represents the components along the respective axes.
+
+        ``micromagneticdata.Drive.info`` is returned as the output ``xarray.DataArray``
+        attributes, besides the ones derived from
+        ``discretisedfield.Field.to_xarray()``.
+
+        Returns
+        -------
+        xarray.DataArray
+
+            ``micromagneticdata.Drive`` as ``xarray.DataArray``
+
+        Examples
+        --------
+        1. Drive to DataArray
+
+        >>> import os
+        >>> import micromagneticdata as md
+        ...
+        >>> dirname = dirname=os.path.join(os.path.dirname(__file__),
+        ...                                'tests', 'test_sample')
+        >>> drive = md.Drive(name='system_name', number=0, dirname=dirname)
+        >>> xr_drive = drive.to_xarray()
+        >>> xr_drive
+        <xarray.DataArray 'field' (t: 25, x: 20, y: 10, z: 4, comp: 3)>
+        ...
+
+        2. Magnetization in a cell over time for ``TimeDriver``
+
+        >>> xr_drive.isel(x=2, y=2, z=2)
+        <xarray.DataArray 'field' (t: 25, comp: 3)>
+        ...
+
+        """
+        if self.info["driver"] == "MinDriver":
+            darray = self[0].to_xarray()
+        else:
+            field_darrays = (field.to_xarray() for field in self)
+            darray = xr.concat(field_darrays, dim=self.table.data[self.table.x])
+            darray[self.table.x].attrs["units"] = self.table.units[self.table.x]
+            if self.info["driver"] == "HysteresisDriver":
+                for i in "xyz":
+                    darray = darray.assign_coords(
+                        {
+                            f"B{i}_hysteresis": (
+                                "B_hysteresis",
+                                self.table.data[f"B{i}_hysteresis"],
+                            )
+                        }
+                    )
+                    darray[f"B{i}_hysteresis"].attrs["units"] = self.table.units[
+                        f"B{i}_hysteresis"
+                    ]
```

Comment:
I am not sure if the issue with a misleading name of `B_hysteresis` has been resolved. In the `table.data` I know that `B_hysteresis` is the norm of the hysteresis vector. Is it the same here, or do we still have and index (counting hysteresis steps)?